### PR TITLE
Add rules for commits to the style guide

### PR DESCRIPTION
**Pull request by lunaryorn**
Because we've been missing these :)

### CODE REVIEW

**Comment by lunaryorn** [comment]
LGTM, @cpitclaudel? @flycheck/contributors?

**Comment by rski** [comment]
Maybe a link to the tpope guidelines (http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) like puppet-mode has? I think it makes the motivation behind good commit messages clear. Unless we don't really adhere to the style rules described by it.

**Comment by lunaryorn** [comment]
@rski Our model commit message is actually a direct inspiration 😎 The post provides a good motivation for the rules, so I guess we should add it as a "see also" link?

**Comment by cpitclaudel** [comment]
LGTM!

**Comment by rski** [comment]
@lunaryorn If you think it adds value, sure. otherwise LGTM

File: doc/contributor/style-guide.rst [hunk 1]
```diff
@@ -117,8 +117,25 @@ Documentation
...
+* Squash trivial fixes into previous commits so that no commit in and by itself
+  violates this style guide.
+
+* Write commit messages that adhere to the style illustrated below.
+
+* In doubt prefer long messages over short messages.  Take the time to write a
+  good message that explains the intention of the change and illustrates
+  noteworthy aspects of the implementation.
+
+* If the a commit fixes a bug try to reproduce a brief description of the bug in
```
Comment (lunaryorn) [inline doc/contributor/style-guide.rst]: Fixed

**Comment by lunaryorn** [comment]
@rski I added a link to the blog post.   Thanks for bringing it up!

Waiting for the Travis build, and will merge afterwards.

**Pull request merged**